variantlib/configuration.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import logging
+import sys
+from enum import IntEnum
+from functools import cache
+from functools import wraps
+from pathlib import Path
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Callable
+from typing import TypeVar
+
+import platformdirs
+
+from variantlib import errors
+from variantlib.constants import CONFIG_FILENAME
+from variantlib.models.configuration import VariantConfiguration as ConfigurationModel
+from variantlib.utils import classproperty
+
+if TYPE_CHECKING:
+    from variantlib.models.variant import VariantFeature
+    from variantlib.models.variant import VariantProperty
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+
+    import tomllib
+else:
+    import tomli as tomllib
+    from typing_extensions import Self
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigEnvironments(IntEnum):
+    LOCAL = 1
+    VIRTUALENV = 2
+    USER = 3
+    GLOBAL = 4
+
+
+@cache
+def get_configuration_files() -> dict[ConfigEnvironments, Path]:
```

Comment:
I think it'd be good to add a test for the values returned here. We could mock `Path.cwd()` and `sys.prefix` to get stable values, and we should be able to affect platformdirs on non-Windows via setting `XDG_CONFIG_DIRS` etc.

Reply:
I'm honestly not so sure what you mean with `we should be able to affect platformdirs on non-Windows via setting XDG_CONFIG_DIRS`

I copied the logic from `pip` - nothing more / nothing less.

For the tests - I may let you add it in a following PR - not exactly sure what you have in mind

Reply:
I mean a test that would verify whether we are getting expected paths. Like if we mock user's home to be `/home/user`, then we should get `/home/user/.config/...` and so on.